LayoutTests/http/tests/security/cross-origin-session-storage-third-party-partitioned.html

```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <script>
+        const testResourcePath = "/security/resources/cross-origin-main-frame-for-partitioned-session-storage.html";
+        const testValue = "1234";
+        let frames = 2;
+        let testWindow;
+        let localhostWindowTestCompleted = false;
+
+        // Step:
+        //   0) Initial load.
+        //   1) Verify stored item in main frame is accessible in localhost subframe.
+        //   2) Verify stored item in main frame is not accessible in loopback subframe.
+        //   3) Store new item in subframe and verify it is accessible in localhost main frame.
+        //   4) Store new item in subframe and verify it is not accessible in loopback main frame.
+        //   5) Stored item in localhost subframe.
+        //   6) Stored item in loopback subframe.
+        //   7) Execute navigation to loopback URL in localhost window.
+        //   8) Verify item stored by localhost subframe is not accessible in main frame.
+        //   9) Verify item stored by loopback subframe is not accessible in main frame.
+        //  10) Return main frame to localhost.
+        //  11) Verify stored item is accessible in localhost subframe.
+        //  12) Verify stored item is accessible in loopback subframe.
+        var currentStep = 0;
+
+        if (window.testRunner) {
+            testRunner.dumpAsText();
+            testRunner.waitUntilDone();
+        }
+
+        function log(line) {
```

Comment:
Not sure if you know this: we have some helper functions in LayoutTests/http/tests/resources/js-test-pre.js, for example you can use debug() for logging, and shouldBeEqualToString() to compare two strings, testPassed()/testFailed for printing something when result is expected/unexpected, etc. I think they will help make the failed case more obvious.

Reply:
I agree it would be nice to write the test with either testharness.js or js-test.js.

Note that js-test-pre.js / js-test-post.js are no longer recommended.

Reply:
Done.